File: 58.cpp
#include <iostream>
#include <math.h>

using namespace std;

void *diag(long n, int ds, int *l)
{
    for (long i = 0; i < n; i++)
    {
        l[i + 1] = l[i] + ds;
        ds += 8;
    cout<<i;
    }
}

void parr(int *arr)
{
    cout << "\n";
    for (int i = 0; i <= sizeof(arr); i++)
        cout << arr[i] << ", ";
    cout << "\n";
}

bool isPrime(long n)
{
    for (int i = 2; i < ceil(sqrt(n)); i++)
        if (!(n % i))
            return 0;
    return 1;
}

int main()
{
    int l[400];
    l[0]= 1;
    diag(11, 8, l);
    parr(l);
    cout << sizeof(l);

    return 0;
}

File: 58.py
from math import sqrt, ceil

def dgnl(n, ds):
    l = [1]
    for i in range(n-1):
        l.append(l[i] + ds)
        ds += 8
    return l

def is_prime(n):
    for i in range(2, ceil(sqrt(n))):
        if not n%i:
            return False
    return True

def all_d(n):
    return set(dgnl(n, 2) + dgnl(n, 4) + dgnl(n, 6)) - {1}


i=2
while True:
    np=len(set(filter(is_prime, all_d(i))))
    tot=(i-1)*4+1
    per = np/tot*100
    print(per)
    if per < 10:
        print(i)
        break
    i+=1

print(dgnl(4,3))